### ISSUE TITLE
After applying commit 6a3a9da, the training job runs out of GPU memory (OOM).

### ISSUE THREAD

**Issue by limou102** [pytorch/torchtitan#1855]
### Bug description

My training job (Flux-dev model) on AMD GPUs encounters an out-of-memory (OOM) issue after merging the following commit from the main branch:
commit id : 6a3a9da9564d82a1120c7639ef6236bb4cffa049
[Refactor attention and make attention mask an argument to the model](https://github.com/pytorch/torchtitan/commit/6a3a9da9564d82a1120c7639ef6236bb4cffa049)
related PR : [Refactor attention and make attention mask an argument to the model](https://github.com/pytorch/torchtitan/pull/1776)

<img width="3300" height="106" alt="Image" src="https://github.com/user-attachments/assets/e50f8ca7-0094-46ae-9f24-4d8310728666" />

Reverting this commit resolves the problem, so it’s possible that this change introduces additional GPU memory usage ?

### Versions

torchtitan commit id : 6a3a9da9564d82a1120c7639ef6236bb4cffa049
torch : 2.10.0.dev20250914+rocm6.4

**Comment by fegin** [comment]
I'm surprised Flux is affected as it doesn't use FlexAttention. Can I get the command you use? Is this specific for AMD GPUs? Also how many steps have you ran? Or you cannot even run the first step?

**Comment by fegin** [comment]
I checked the code, Flux doesn't seem to use attention.py and has its own train.py. So Flux shouldn't be affected by the refactor. @wwwjn is my understanding correct? Or do I miss anything?

**Comment by tianyu-l** [comment]
oh I think the bug was introduced here -- now with wrong indentation
https://github.com/pytorch/torchtitan/pull/1776/files#diff-83b7868cc3b5fde38ae75ccd8346675495ed27207bc75c422cf8c2ef4d8096d3L210-L218

**Comment by wwwjn** [comment]
> oh I think the bug was introduced here -- now with wrong indentation https://github.com/pytorch/torchtitan/pull/1776/files#diff-83b7868cc3b5fde38ae75ccd8346675495ed27207bc75c422cf8c2ef4d8096d3L210-L218

Can you elaborate more on this? Why this causing memory usage increase?

**Comment by fegin** [comment]
I think it is a different issue. Flux does not support CP yet.
